controllers/datadogagent/component/agent/default.go

```diff
@@ -260,11 +260,23 @@ func envVarsForSecurityAgent(dda metav1.Object) []corev1.EnvVar {
 	return append(envs, commonEnvVars(dda)...)
 }
 
+func volumeMountsForInitConfig() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		component.GetVolumeMountForLogs(),
+		component.GetVolumeMountForChecksd(),
+		component.GetVolumeMountForConfd(),
+		component.GetVolumeMountForConfig(),
+		component.GetVolumeMountForProc(),
+		component.GetVolumeMountForRuntimeSocket(true),
```

Comment:
do we know why we need so many volume in the init container?

Reply:
i'm not sure the reasons for all of them but i matched what is present in a helm deployment